### PR TITLE
Only small and tiny simplemobs can go on skateboards, shoot people off them [BALANCE PRE APPROVED]

### PR DESCRIPTION
**Pull request by Qwertytoforty**
Sorry for doing a balance during the freeze already, but I'm not letting this be a complaint for a month. Balance pre approved.

## What Does This PR Do

Simple mob animals that are normal or larger sized, can not ride skateboards.
The only hostile mob that can ride a skateboard that is a simple mob, is viscerators, and headslugs.
Cyborgs, humans, monkeys, and xenomorphs can still ride them. AI sadly can not.

You can now knock people off skateboards, unbuckling them and knocking them down for 2 seconds, by hitting them with a projectile when they have 60 or more stamina damage, or 60 or more damage. If the user is stun immune (his grace, hulk, stimulants), this will not work

This means skateboards while still effective and high risk, don't let you endlessly evade. Eventually projectiles will catch up to you and get you off it.

## Why It's Good For The Game

Ok, while funny terrors and juggernaughts on skateboard passed the funny balance of say a ambulance, and became a balance nightmare with their speed.
This blocks it for skateboards.
Janitor carts and such unaffected, as they are slower vehicles.

Skateboards were too effective at getaway, this should mean officers can still down those on skateboards one way or another

## Testing

Shot skrell on skateboards with lethals and non lethals, to see them fall off.

Rode a skateboard as a human.
Rode a skateboard as a xenomorph.
Rode a skateboard as a cyborg.
Rode a skateboard as a bunny (size 0
Rode a skateboard as a fox (size 1)
Failed to ride a skateboard as a spider (size 2)

## Changelog
:cl:
tweak: Normal and large simple animals can no longer ride skateboards.
tweak: A person with 60 or more damage or stamina damage, can be shot off a skateboard.
/:cl:

### CODE REVIEW

**Comment by Komrad822** [comment]
I REALLY wish I had saved a recording of the juggs on skateboards.... The sheer terror as they rolled up.

**Review by Fordoxia — APPROVED** [review]
Code looks good to me. Being able to shoot people off skateboards is also very nice.

File: code/datums/components/riding/riding_vehicle.dm [hunk 1]
```diff
@@ -123,23 +123,37 @@
...
+	SIGNAL_HANDLER // COMSIG_ATOM_BULLET_ACT
+	if(!istype(parent, /obj/tgvehicle/scooter/skateboard))
+		return
+	var/obj/tgvehicle/scooter/skateboard/S = parent
+	for(var/mob/living/L in S.return_occupants()) // Only one on a skateboard unless an admin var edits it. If an admin var edits it, that is on them.
+		if((L.staminaloss >= 60 || L.health <= 40) && !L.absorb_stun(0)) // Only injured people can be shot off. Hulks and people on stimulants can not be shot off.
+			S.unbuckle_mob(L)
+			L.KnockDown(2 SECONDS)
+			L.visible_message("<span class='warning'>[L] gets shot off [S] by [projectile]!</span>",
+			"<span class='warning'>You get shot off [S] by [projectile]!</span>")
```
Comment (Sirryan2002) [inline code/datums/components/riding/riding_vehicle.dm:151]:
```suggestion
				"<span class='warning'>You get shot off [S] by [projectile]!</span>")
```
indentation

File: code/datums/components/riding/riding_vehicle.dm [hunk 2]
```diff
@@ -123,23 +123,37 @@
...
 		RegisterSignal(rider, COMSIG_MOVE_INTENT_TOGGLED, PROC_REF(toggle_move_delay))
 		toggle_move_delay(rider)
+		RegisterSignal(rider, COMSIG_ATOM_BULLET_ACT, PROC_REF(check_knockoff))
 
 /datum/component/riding/vehicle/scooter/skateboard/handle_unbuckle(mob/living/rider)
 	. = ..()
 	if(can_slow_down)
 		toggle_move_delay(rider)
 		UnregisterSignal(rider, COMSIG_MOVE_INTENT_TOGGLED)
+	UnregisterSignal(rider, COMSIG_ATOM_BULLET_ACT)
```
Comment (Sirryan2002) [inline code/datums/components/riding/riding_vehicle.dm:133]: is this meant to be outside this `if` block?

Comment (Qwertytoforty) [reply code/datums/components/riding/riding_vehicle.dm:133]: Yes, as it is not applied inside can_slow_down- ok
It is applied in can slow down, it should not be. Will remove it from inside the other if block, thanks for spotting!

**Review by Sirryan2002 — APPROVED** [review]

**Review by lewcc — APPROVED** [review]

**Pull request merged**